src/docfx/build/toc/parser/MarkdownTocMarkup.cs

```diff
@@ -94,43 +95,36 @@ private static (List<TableOfContentsInputItem> children, int count) ConvertTo(st
 
             TableOfContentsInputItem GetItem(HeadingBlock block)
             {
+                var currentItem = new TableOfContentsInputItem();
                 if (block.Inline == null || !block.Inline.Any())
                 {
                     errors.Add(Errors.MissingTocHead(new Range(block.Line, block.Column), filePath));
-                    return new TableOfContentsInputItem();
+                    return currentItem;
+                }
+
+                var xrefLink = block.Inline.FirstOrDefault(l => l is XrefInline);
+                if (xrefLink != null && xrefLink is XrefInline xrefInline && !string.IsNullOrEmpty(xrefInline.Href))
+                {
+                    currentItem.Uid = xrefInline.Href;
+                    return currentItem;
                 }
 
-                string name = null;
-                string displayName = null;
-                string href = null;
                 var link = block.Inline.FirstOrDefault(l => l is LinkInline);
+                var literalLink = block.Inline.FirstOrDefault(l => l is LiteralInline);
```

Comment:
Probably not related to this PR but `FirstOrDefault` might be too permissive, when there are more inline elements, it abandoned the remaining ones without a warning. Same is true for `LinkInline` and `LiteralInline` handling

```
# @b xyz
```

Reply:
yes, i remember we talked about that, will test it later and create an new pr if we need change it.